code/search/SearchIndex.php

```diff
@@ -77,26 +98,49 @@ public function fieldData($field, $forceType = null, $extraOptions = array())
             foreach ($lookups as $lookup) {
                 $next = array();
 
-                foreach ($sources as $source => $options) {
-                    $class = null;
+                foreach ($sources as $source => $baseOptions) {
```

Comment:
Please update the `fieldData()` method docs to note on how it handles relationships